enhancements/fail-forward-upgrades.md

```diff
@@ -0,0 +1,252 @@
+- As a catalog curator / operator author, I want to be able to prevent a specific version of the operator from "failing forward" as it might include a critical update such as a custom resource migration.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Opting into Fail Forward Upgrades
+
+Since resolution is namespaced, the toggle for allowing "fail forward" upgrades should be namespaced as well to avoid having to add the extra burden on the resolver surrounding understanding which dependent operators can or cannot use the forced upgrade toggle.
+
+With this in mind, this enhancement proposes adding the `upgradeStrategy` field to an already watched namespaced resource, the OperatorGroup.
+
+```yaml
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: foo
+  namespace: bar
+spec:
+  upgradeStrategy:
+     # Possible values include "default" or "failForward".
+    type: failForward
+```
+
+With the the upgradeStrategy is set to `failForward`, OLM will allow operators to "fail forward" in adherence with the principles discussed below. If the upgradeStrategy is unset or set to `default` OLM will exhibit existing behavior.
+
+#### Supporting "Fail Forward" Upgrades
+
+Before diving into the specific behavior OLM will need to support when "falling forward" from failed CSVs or InstallPlans, it is important to understand how OLM and the Resolver calculate what needs to be installed. When determining what to install, OLM provides the resolver with the set of CSVs in a namespace. The resolver treats these CSVs differently depending on whether or not they are claimed by a Subscription, where the Subscription lists the CSV's name in its `.status.currentCSV` field.
+- If a CSV is claimed by a Subscription, the resolver will allow it to be upgraded by a bundle that replaces it.
+- If a CSV is not claimed by a Subscription, it is assumed that the user installed the CSV themselves, that OLM should not upgrade the CSV, and that it must appear in the next set of installables.
+
+OLM's Resolver is deterministic, meaning that the set of installables will not change for a given set of arguments if no new upgrades have been declared in the existing CatalogSources. When an upgrade fails due to the new CSV entering the FAILED phase, the csv being replaced still exists and is not referenced by a subscription. Today, OLM would send both `Operator v1` and `Operator v2` to the resolver which will normally be unsatisifiable because `Operator v1` is marked as required by the resolver and `Operator v2` cannot be upgraded further. In order to support "failing forward", OLM will need to omit CSVs that are being replaced by an operator from the set of arguments sent to the resolver when "fail forward" upgrades are enabled.
+
+##### Supporting "Fail Forward" Upgrades when one or more CSVs have failed
```

Comment:
Seems like this heading is about CSVs, but the body is about Install Plans? And vice versa for the below heading/body.

I think just swap the headings or bodies?